src/Guzzle/SubjectRewritingMiddleware.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace eLife\Journal\Guzzle;
+
+use GuzzleHttp\Promise\PromiseInterface;
+use GuzzleHttp\Psr7\Uri;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+use function GuzzleHttp\json_decode;
+use function GuzzleHttp\Psr7\parse_query;
+use function GuzzleHttp\Psr7\stream_for;
+
+final class SubjectRewritingMiddleware
+{
+    private $replacements = [
+        // Dummy rewrite
+        'cancer-biology' => [
+            'id' => 'immunology',
+            'name' => 'Immunology',
+        ],
+    ];
+
+    public function __invoke(callable $handler)
+    {
+        return function (RequestInterface $request, array $options) use (&$handler) {
+            $uri = $request->getUri();
+
+            foreach ($this->replacements as $replaced => $replacement) {
+                if (in_array($uri->getPath(), ["/highlights/{$replaced}", "/subjects/{$replaced}"])) {
+                    $uri = $uri->withPath("/subjects/{$replacement['id']}");
+                }
+            }
+
+            $query = parse_query($request->getUri()->getQuery());
+            if (!empty($query['subject[]'])) {
+                $subjects = array_flip((array) $query['subject[]']);
+
+                foreach ($this->replacements as $replaced => $replacement) {
+                    if (isset($subjects[$replaced])) {
+                        unset($subjects[$replaced]);
+                        $subjects[$replacement['id']] = null;
+                    }
+                }
+
+                $uri = Uri::withoutQueryValue($uri, 'subject[]');
+                foreach (array_keys($subjects) as $subject) {
+                    $uri = Uri::withQueryValue($uri, 'subject[]', $subject);
+                }
+            }
```

Comment:
replacing subjects in paths and in query string look generic enough to be extracted into a couple of private methods